kernel/src/engine/parquet_stats_skipping.rs

```diff
@@ -0,0 +1,1267 @@
+                    Some(self.get_nullcount_stat_value(&col)? != skip)
+                }
+                _ => None,
+            },
+        }
+    }
+
+    /// Propagates a boolean-typed column, allowing e.g. `flag OR ...`.
+    /// Columns of other types are ignored (NULL result).
+    fn apply_column(&self, col: &str, inverted: bool) -> Option<bool> {
+        let col = col_name_to_path(col);
+        let as_boolean = |get: &dyn Fn(_, _, _) -> _| match get(self, &col, &DataType::BOOLEAN) {
+            Some(Scalar::Boolean(value)) => Some(value),
+            _ => None,
+        };
+        let min = as_boolean(&Self::get_min_stat_value)?;
+        let max = as_boolean(&Self::get_max_stat_value)?;
+        Some(min != inverted || max != inverted)
+    }
+
+    /// Propagates a boolean literal, allowing e.g. `FALSE OR ...`.
+    /// Literals of other types are ignored (NULL result).
+    fn apply_scalar(value: &Scalar, inverted: bool) -> Option<bool> {
+        match value {
+            Scalar::Boolean(value) => Some(*value != inverted),
+            _ => None,
+        }
+    }
+
+    /// Performs a partial comparison against a column min-stat. See [`partial_cmp_scalars`] for
+    /// details of the comparison semantics.
+    fn partial_cmp_min_stat(
+        &self,
+        col: &ColumnPath,
+        val: &Scalar,
+        ord: Ordering,
+        inverted: bool,
+    ) -> Option<bool> {
+        let min = self.get_min_stat_value(col, &val.data_type())?;
+        partial_cmp_scalars(&min, val, ord, inverted)
+    }
+
+    /// Performs a partial comparison against a column max-stat. See [`partial_cmp_scalars`] for
+    /// details of the comparison semantics.
+    fn partial_cmp_max_stat(
+        &self,
+        col: &ColumnPath,
+        val: &Scalar,
+        ord: Ordering,
+        inverted: bool,
+    ) -> Option<bool> {
+        let max = self.get_max_stat_value(col, &val.data_type())?;
+        partial_cmp_scalars(&max, val, ord, inverted)
+    }
+}
+
+/// Compares two scalar values, returning Some(true) if the result matches the target `Ordering`. If
+/// an inverted comparison is requested, then return Some(false) on match instead. For example,
+/// requesting an inverted `Ordering::Less` matches both `Ordering::Greater` and `Ordering::Equal`,
+/// corresponding to a logical `>=` comparison. Returns `None` if the values are incomparable, which
+/// can occur because the types differ or because the type itself is incomparable.
+pub(crate) fn partial_cmp_scalars(
+    a: &Scalar,
+    b: &Scalar,
+    ord: Ordering,
+    inverted: bool,
+) -> Option<bool> {
+    let result = a.partial_cmp(b)? == ord;
+    Some(result != inverted)
+}
+
+pub(crate) fn col_name_to_path(col: &str) -> ColumnPath {
+    // TODO: properly handle nested columns
+    // https://github.com/delta-incubator/delta-kernel-rs/issues/86
+    ColumnPath::new(col.split('.').map(|s| s.to_string()).collect())
+}
+
+#[cfg(test)]
```

Comment:
given the size of these tests, i think it might be best to make the file tree:
```
kernel/src/engine/
  ...
  parquet_stats_skipping.rs
  parquet_stats_skipping/
    tests.rs
```

And then here you can just do:
```rust
#[cfg(test)]
mod tests;
```

Reply:
ah nice... I wanted to do something like that but wasn't sure how.